arch/arm/src/common/Toolchain.defs

```diff
@@ -267,6 +254,29 @@ endif
 
 ARCHOPTIMIZATION += -fno-common -Wall -Wshadow -Wundef
 
+ifeq ($(CONFIG_ARM_THUMB),y)
+  ARCHOPTIMIZATION += -mthumb
+
+  # Add assembler options only if GCC LTO is not enabled. Otherwise we get
+  # lto-wrapper: warning: Extra option to '-Xassembler': -mthumb, dropping
+  # all '-Xassembler' and '-Wa' options.
+
+  ifneq ($(LD),$(CC))
```

Comment:
Need to find another solution as `ARCHOPTIMIZATION` is passed to both `CC` and `LD`, but `-Wa` should be filtered only for `LD`.

Reply:
Maybe `ARCHOPTIMIZATION` should not be passed to LD

Reply:
Seems not to be possible because of
```
    LDSTARTGROUP ?= -Wl,--start-group
    LDENDGROUP   ?= -Wl,--end-group
    LDFLAGS      := $(addprefix -Xlinker ,$(LDFLAGS))
    LDFLAGS      += $(CFLAGS)
```

Reply:
How about ignore this warning? -Werror doesn't care about warning from lto-wrapper, it looks more like a prompt message